src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -304,6 +304,9 @@ private static bool OverloadResolutionResultIsValid<TMember>(ArrayBuilder<Member
                 RemoveLessDerivedMembers(results, ref useSiteInfo);
             }
 
+            // PROTOTYPE(instance) The preference for more specific extension members should also apply to classic extension methods (without affecting back compat, somehow).
+            RemoveLessSpecificExtensionMembers(results, ref useSiteInfo);
```

Comment:
>RemoveLessSpecificExtensionMembers [](http://example.com/codeflow?start=12&length=34)

It feels like the call should be placed into the previous `if`.

Reply:
That's something I'd looked at, but I don't think we should.
Here's the logic driving that condition:
```
            // No need to check for overridden or hidden methods if the members were
            // resolved as extension methods and the extension methods are defined in
            // types derived from System.Object.
            bool checkOverriddenOrHidden = !(isExtensionMethodResolution &&
                members.All(static m => m.ContainingSymbol is NamedTypeSymbol { BaseTypeNoUseSiteDiagnostics.SpecialType: SpecialType.System_Object }));

```

I'd be open to adding a separate flag to indicate that we have no extension members, so this step could be skipped, as an optimization. Let me know what you think

Reply:
> I'd be open to adding a separate flag to indicate that we have no extension members, so this step could be skipped, as an optimization. Let me know what you think

Perhaps a comment?

Reply:
On second thought, the cost of calling `RemoveLessSpecificExtensionMembers` when there are no extension members should be the same as the check for initializing a flag (scan the results once and check whether the containing types are extensions).
I'll leave as-is.

Reply:
> On second thought, the cost of calling RemoveLessSpecificExtensionMembers when there are no extension members should be the same as the check for initializing a flag (scan the results once and check whether the containing types are extensions).
I'll leave as-is.

Is the following an accurate description of the situation?
1. We set ```checkOverriddenOrHidden``` to false as an optimization measure and the usage of the flag is not meant to affect the end result
2. We do not expect to get any noticeable perf gains from making this check conditional

If so, consider adding a comment explaining that.

Reply:
Yes, that's my understanding.
I think it's okay as-is. Thanks